app/presigned_urls.py

```diff
@@ -0,0 +1,89 @@
+import json
+
+import logging
+import boto3
+
+logger = logging.getLogger()
+
+EXPIRATION_TIME = 3600  # 3600 seconds == 1 hour
+UPLOAD = 'upload'
+DOWNLOAD = 'download'
+
+
+def make_upload_url(full_object_path: str, bucket: str) -> dict:
+    s3 = boto3.client("s3")
+    return s3.generate_presigned_post(
+        bucket, full_object_path, Fields=None, Conditions=None, ExpiresIn=EXPIRATION_TIME
+    )
+
+
+def make_download_url(full_object_path: str, bucket: str) -> dict:
+    s3 = boto3.client("s3")
+    url = s3.generate_presigned_url(
+        "get_object",
+        Params={"Bucket": bucket, "Key": full_object_path},
+        ExpiresIn=EXPIRATION_TIME,
+    )
+    return {
+        "url": url
+    }
+
+
+url_makers = {
+    UPLOAD: make_upload_url,
+    DOWNLOAD: make_download_url
+}
+
+
+def make_presigned_url(event, _context, _kwargs):
+    bucket_name = 'garden-mlflow-models-dev'
+    payload = json.loads(event["body"])
+    s3_path = payload.get("s3_path", None)
+    direction = payload.get("direction", None)
+
+    if direction not in url_makers:
+        return {
+            "statusCode": 400,
+            "body": json.dumps({"message": f"'direction' must be one of {UPLOAD} or {DOWNLOAD}. Got {direction}"}),
+        }
+
+    if s3_path is None or not is_probably_valid_object_path(s3_path):
+        message = "'s3_path' must be formatted like '<email address>/<model short name>/model.zip'. "
+        message += f"Got {s3_path}"
+        return {
+            "statusCode": 400,
+            "body": json.dumps({"message": message}),
+        }
+
+    make_url = url_makers[direction]
+    try:
+        url_and_fields_payload = make_url(s3_path, bucket_name)
+    except Exception as e:
+        return {"statusCode": 500, "body": str(e)}
+
+    return {
+        "statusCode": 200,
+        "body": json.dumps(url_and_fields_payload),
+    }
+
+
+def is_probably_valid_object_path(object_path: str):
```

Comment:
Just to make sure that I'm on the same page, we don't really mind if something less-obviously malformed slips through because we think it's unlikely, or because it's fine to always just return 500 plus whatever exception message from the s3 client anyways?

Reply:
I'm of a mind right now where I'm letting our backend be "dumb pipes" and have the serious business logic live in the CLI. When we get to the "real" backend I want to invert that and have our serious validation on the backend. But yeah for now the reasoning is that I'm trusting the CLI to do its job so I think something malformed slipping through is unlikely. (And if it's super badly malformed then S3 would rebel)